### ISSUE TITLE
Add custom Auto-Type shortcuts 

### ISSUE THREAD

**Issue by droidmonkey** [keepassxreboot/keepassxc#5962]
Perhaps an alternative would be to allow the user to customize them if they decide they want them. Something like the following:
![auto-type](https://user-images.githubusercontent.com/48502913/104963220-abaf5a00-599f-11eb-9749-cf9648f069a9.png)

_Originally posted by @mekaneck in https://github.com/keepassxreboot/keepassxc/issues/2093#issuecomment-762473093_

**Comment by ccoenen** [comment]
Can there be `n` (or at least 3)? TOTP is now common enough that this would drive me crazy if it wasn't possible to configure alongside username and password.

**Comment by mekaneck** [comment]
I'll use my amazing MS Paint skills once more for a refined suggestion. To accommodate more customization without disturbing users who don't desire it, why not pull the existing global auto-type shortcut into a customizable one? The default settings for it can be made the same as what it is today (User tab PW enter) and carry over whatever the user has for a shortcut key. That way, upon upgrade, there is no changes to them. However, if someone (for example) doesn't like the {ENTER} at the end of the sequence, they can take it off if they wish without wasting one of the other hotkeys. 

I also added an additional shortcut to accommodate TOTP as suggested by @ccoenen . My MS Paint skills aren't good enough to show what `n` hotkeys might look like in the GUI, so this is the best I can do. 

![image](https://user-images.githubusercontent.com/48502913/105609940-25fe2680-5d72-11eb-9525-890518b75a5e.png)

**Comment by droidmonkey** [comment]
No need for a mock up I can see the vision already

**Comment by hifi** [comment]
Is this needed after the Auto-Type PR with enhanced dialog is merged?

This creates a shortcut but should we first see how people take in the new dialog?

**Comment by droidmonkey** [comment]
Yah I want to add this afterwards on with a "Summon KPXC" global shortcut

**Comment by Ferk** [comment]
The "Summon KPXC" (eg. open the main window and bring it up front) is something I've been missing from the original Keepass.
It'd be great if that gets added.

In theory that could be used in a shortcut chain to do more things (so you first Summon the main window and immediately follow up with an already existing non-global shortcut from there).

**Comment by murder** [comment]
Guys,

Been a while - any news? At least for ()OTP ?

**Comment by vphantom** [comment]
By the looks of the milestones, v2.7.2 didn't happen but it could still be on track for v2.8.0. So many web sites break password managers on purpose (first page for username, second page for password, ever-changing CAPTCHA sequence, etc.) that it will be quite a relief for me to be able to create 3 global shortcuts (username, password, OTP) to navigate those hostile sites.

**Comment by murder** [comment]
> By the looks of the milestones, v2.7.2 didn't happen but it could still be on track for v2.8.0. So many web sites break password managers on purpose (first page for username, second page for password, ever-changing CAPTCHA sequence, etc.) that it will be quite a relief for me to be able to create 3 global shortcuts (username, password, OTP) to navigate those hostile sites.

The idea is to never {ENTER}

**Comment by vphantom** [comment]
Skipping `{ENTER}` is not enough: on the more hostile sites, the user needs to manually click some kind of submit button which remains disabled until the username and sometimes even the CAPTCHA is done. They're basically begging users to use `123456` instead of password managers, IMHO, but hey all in the name of perceived security…

**Comment by theincogtion** [comment]
May also solve this request:
https://github.com/keepassxreboot/keepassxc/issues/8999

**Comment by PoisonFrog** [comment]
I'm already used to the Ctrl +1, Ctrl +2 and Ctrl +3 for username, password and TOTP respectively, but I wish I could add an additional attribute to Ctrl + 4. For example, let's say I have some entries with "app-password" as an additional attribute, would it be possible to add this additional attribute to the global auto-type with this change?

**Comment by droidmonkey** [comment]
Interesting idea, shouldn't be too hard.

**Comment by khurozov** [comment]
Hi. I've recently migrated to KeePassXC. I was using custom bash script around `pass`. I liked "Type Sequence" dropdown  in Auto-type dialog very much. 

Only thing I'm missing is typing URLs as I have long long exact locations for some passwords.

Could you please add another dropdown item for URL with hot key, for example Ctrl+5.

I would really appreciate it.

**Comment by PoisonFrog** [comment]
> Hi. I've recently migrated to KeePassXC. I was using custom bash script around `pass`. I liked "Type Sequence" dropdown in Auto-type dialog very much.
> 
> Only thing I'm missing is typing URLs as I have long long exact locations for some passwords.
> 
> Could you please add another dropdown item for URL with hot key, for example Ctrl+5.
> 
> I would really appreciate it.

Seems like you're trying to use KeePassXC Auto-Type entries as your web bookmarks.

**Comment by droidmonkey** [comment]
> Only thing I'm missing is typing URLs as I have long long exact locations for some passwords.

You can just click the url in the entry preview.

**Comment by khurozov** [comment]
> > Hi. I've recently migrated to KeePassXC. I was using custom bash script around `pass`. I liked "Type Sequence" dropdown in Auto-type dialog very much.
> > Only thing I'm missing is typing URLs as I have long long exact locations for some passwords.
> > Could you please add another dropdown item for URL with hot key, for example Ctrl+5.
> > I would really appreciate it.
> 
> Seems like you're trying to use KeePassXC Auto-Type entries as your web bookmarks.

Yes, I am. Because I use 3 browsers and was autotyping urls with script combining pass, dmenu and xdotool. It was easy to use and I had a hotkey for complete functionalities I wanted.

**Comment by khurozov** [comment]
> > Only thing I'm missing is typing URLs as I have long long exact locations for some passwords.
> 
> You can just click the url in the entry preview.

Isn't there any workaround to type URL without opening main window?

**Comment by droidmonkey** [comment]
Yah you can define an auto type sequence for the entry of {URL}. Then choose that sequence from the autotype selection dialog.

**Comment by khurozov** [comment]
ok, thanks

**Comment by Grundik** [comment]
That would be helpful, if at least predefined list of auto-type sequences would have been editable. Even if that would be only manually accessible though the entry menu (like its already are for most of items). That could be done, for example, by putting all custom window sequences into that menu.

<img width="539" height="221" alt="Image" src="https://github.com/user-attachments/assets/a54ed86f-2de5-4415-a06e-9edd7b2e2c29" />

**Comment by Chealer** [comment]
@mekaneck: Your proposal is interesting, but could result in considerable shortcut pollution. Shortcuts are becoming a hot commodity on Windows. It's amazing to realize everything your system already supports when you try crafting a new one.😁

It may be wiser to have something between your proposal and @Grundik’s. Perhaps offer something like @Grundik’s menu via the tray menu, and offer a shortcut to raise KeePassXC’s tray menu. Possibly 1 extra keystroke, but less pollution and memorization. Ticket #149 already suggests adding “Perform Auto-Type“ in the tray menu, which would be a good start, although there could be preferences to customize the sequences offered.

<details>
  <summary>License</summary>
All comments and attachments from myself on this ticket are offered under the terms of CC0 1.0.
</details>

**Comment by Gittyperson** [comment]
I use CTRL+1 and CTRL+2 a lot (especially the latter). Can someone please explain why ENTER is not included as default? Is there a way to reverse this i.e. have CTRL+2 include ENTER?

Could be just me, but every single time I use CTRL+2, manually pressing Enter is required afterwards. Not the end of the world, but it's a 100% scenario with no exceptions.

**Comment by droidmonkey** [comment]
Quite simply so you can control the process flow. Type username, then type password. Type Password then TOTP. If you want enter to be pressed then define a sequence...

**Comment by Gittyperson** [comment]
If I understand this correctly, it would mean to define an extra PASS+ENTER sequence for every single Auto-Type login entry? If so, it's not really realistic or even possible with hundreds of entries. Unless I am missing something?

For me at least, the usual scenario is when websites keep you "half-logged in". They've stored the username but want the password entered again. It's then time for CTRL+2+ENTER.

**Comment by ccoenen** [comment]
Personally, I do not want <kbd>enter</kbd> to be included in <kbd>ctrl+2</kbd> by default.

Imagine a very simple accident where the wrong field has focus. You're now looking at your password in clear text, and you might catch that just in time if you press enter manually.

Also, it's not what I asked KeePassXC to do. I asked it for my password, not for my password PLUS <kbd>ENTER</kbd>. Or PLUS <kbd>TAB</kbd> or PLUS <kbd>Esc</kbd>. Any further combination is context dependent, and KeePass shouldn't guess what is correct.

I might be in the "change password" dialog of a website. The extra <kbd>enter</kbd> is now a hindrance. Maybe the login flow in this particular case has unforseen extra steps. I have a feeling that his is more and more the norm (login forms that change a lot while logging in). I might want to check the "remember me"/"keep me logged in" checkbox ... so many reasons to not change this particular behavior.

**Comment by Gittyperson** [comment]
Fair enough. I am asking for consideration of additional shortcuts for separate user/pass that include ENTER. I believe the scenario I mentioned is common enough, for PASS+ENTER at least.

**Comment by droidmonkey** [comment]
Is pressing enter that taxing? You are already hands on keyboard to type Ctrl+2 🤔 we wont be adding new shortcuts to the autotype selection dialog, the number of existing options is already large and somewhat undiscoverable.
